service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/HmctsEmailContentProvider.java

```diff
@@ -30,4 +54,56 @@ public HmctsEmailContentProvider(LocalAuthorityNameLookupConfiguration localAuth
             .put("localAuthority", localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthorityCode))
             .build();
     }
+
+    public Map<String, Object> buildC2UploadNotification(final CaseDetails caseDetails,
+                                                         final String localAuthorityCode) {
+        // Validation within our frontend ensures that the following data is present
+        CaseData caseData = mapper.getObjectMapper().convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Map<String, Object>> respondents1 =
+            (ObjectUtils.isEmpty(caseDetails.getData().get("respondents1"))
+                ? Collections.emptyList() : mapper.getObjectMapper().convertValue(
+                    caseDetails.getData().get("respondents1"), new TypeReference<>() {}));
+
+        List<Respondent> respondents = (CollectionUtils.isEmpty(respondents1)
+            ? Collections.emptyList() : respondents1.stream().map(
+                respondent -> mapper.getObjectMapper().convertValue(
+                    respondent.get("value"), Respondent.class)).collect(toList()));
+
+        return super.getCasePersonalisationBuilder(caseDetails)
+            .put("court", hmctsCourtLookupConfiguration.getCourt(localAuthorityCode).getName())
+            .put("lastNameOfRespondent", getRespondent1Lastname(respondents))
+            .put("familyManCaseNumber", StringUtils.defaultIfBlank(caseData.getFamilyManCaseNumber(), ""))
+            .put("hearingDate", dateFormatterService.formatLocalDateToString(
+                getHearingBookingDate(caseData), FormatStyle.MEDIUM))
+            .build();
+    }
+
+    private LocalDate getHearingBookingDate(final CaseData caseData) {
```

Comment:
are you able to use the get most urgent hearing booking method in hearing booking service class?

Reply:
thanks for the spot @toireasam, this field has now been taken out.